### ISSUE TITLE
Docker images used for database-backed tests are not updated

### ISSUE THREAD

**Issue by StevenMaude** [opensafely-core/databuilder#722]
```
tests/lib/databases.py
161:        image="mcr.microsoft.com/mssql/server:2017-CU25-ubuntu-16.04",
194:            image="docker.io/bitnami/spark:3.1.2-debian-10-r126",
```

* The [Spark version is a year old](https://hub.docker.com/layers/bitnami/spark/3.1.2-debian-10-r126/images/sha256-8803352556379fa9faff8cae794a6bf635359e89bb26421cdb40ee9703fe088b?context=explore) and using Debian 10, when they have a Debian 11 version.
* The [SQL Server image](https://hub.docker.com/_/microsoft-mssql-server?tab=description) is also over a year old, and using Ubuntu 16.04. They do still have images from this year, but I'm not sure how long they'll continue to build with 16.04; it's end-of-life unless you pay Canonical for extended support. 

Ideally, we'd have a way of checking for updates for these when appropriate. The risks are probably fairly low as we're running trusted code in the tests, and presumably not running the servers in public anywhere. But if we can upgrade, then we should.

**Comment by evansd** [comment]
Thanks @StevenMaude. I agree the risks are pretty low here as these never run exposed to the internet, but it's still better to be up-to-date where we can.

I've made the updates in:
 * #727

For the longer term, I wonder if a scheduled task similar to the one you made [here](https://github.com/opensafely-core/cohort-extractor/blob/de3915e94c8b7ad676b9c072a63ceb10e46508ce/.github/workflows/emis_trino_version_check.yaml) might be the way to keep track of this?

**Comment by StevenMaude** [comment]
> For the longer term, I wonder if a scheduled task similar to the one you made [here](https://github.com/opensafely-core/cohort-extractor/blob/de3915e94c8b7ad676b9c072a63ceb10e46508ce/.github/workflows/emis_trino_version_check.yaml) might be the way to keep track of this?

Yes, this is probably the right thing to do, when there's time :+1:

***

A couple of incidental things that I realised thinking about this; these themselves might be individual issues:

1. We're just using [`2017-latest` in cohort-extractor](https://github.com/opensafely-core/cohort-extractor/blob/de3915e94c8b7ad676b9c072a63ceb10e46508ce/docker-compose.yml#L3), which ensures that the CI runs with the latest version, but doesn't mean developers will. It might warrant pinning too.

2. There's also a related question, which might be a separate issue, of _which_ versions we need to test against. With cohort-extractor, the versions there were chosen to be compatible with whatever TPP and EMIS are running.

   Are we going to specify to data providers that *only* SQL Server 2017 is supported? Or are we going to support a range of versions (e.g. 2017, 2019, 2022)? And, if we *do* intend to support those versions, then do we need to add them to run with the tests; at least, maybe in GitHub Actions CI?

**Comment by benbc** [comment]
@StevenMaude On your point 2, I don't think that we need to worry about this. For now the only backends we need to support are TPP and (at some point EMIS). We can just test exactly the database versions that they use. If we add another one or two new backends we can extend our testing to the databases _they_ use. It's only when we get beyond this that we need to provide generalized support for databases independent of the backends that use them.
